lib/usecase/login_use_case.dart

```diff
@@ -0,0 +1,44 @@
+import 'dart:async';
+
+import 'package:injectable/injectable.dart';
+import 'package:survey/api/repository/auth_repository.dart';
+import 'package:survey/api/response/login_response.dart';
+import 'package:survey/usecase/base/base_use_case.dart';
+import 'package:survey/utils/shared_preferences_utils.dart';
+
+class LoginInput {
+  final String email;
+  final String password;
+
+  LoginInput({
+    required this.email,
+    required this.password,
+  });
+}
+
+@Injectable()
+class LoginUseCase extends UseCase<void, LoginInput> {
+  final AuthRepository _repository;
+  final SharedPreferencesUtils _sharedPreferencesUtils;
+
+  const LoginUseCase(
+    this._repository,
+    this._sharedPreferencesUtils,
+  );
+
+  @override
+  Future<Result<void>> call(LoginInput input) {
```

Comment:
Should this usecase return a `Result<Login>` instead of `Result<void>`?

Reply:
Since we only care if this use case returns success or failed, I'm not sure why we need to return `Result<Login>`. Could you please explain more about this? 🙏🏼

Reply:
@chornerman Ah sorry, for this usecase, we don't need to get its result for any next steps 🙏